### PR TITLE
Added heat pipe microreactor assembly model

### PR DESCRIPTION
**Pull request by joshuahansel**

### CODE REVIEW

**Comment by joshuahansel** [comment]
This will not be able to run until https://github.com/idaholab/moose/pull/25258 is merged and makes it into DireWolf.

**Comment by GiudGiud** [comment]
Do you have a paper to cite for this model?

File: microreactors/hpmr_assembly/main.i [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+  ${x2} ${y2} ${hp_axial_offset}
+  ${x3} ${y3} ${hp_axial_offset}
+  ${x4} ${y4} ${hp_axial_offset}
+  ${x5} ${y5} ${hp_axial_offset}
+  ${x6} ${y6} ${hp_axial_offset}'
+
+[Mesh]
+  [file]
+    type = FileMeshGenerator
+    file = mesh.e
```
Comment (GiudGiud) [inline microreactors/hpmr_assembly/main.i:70]: where did you get the mesh?
should there be a meshing script checked in?

**Comment by joshuahansel** [comment]
I couldn't find your comment, but yes, there is a python script to compute the HP positions. Do you want me to push it?

File: microreactors/hpmr_assembly/tests [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[Tests]
+  [generate_mesh]
+    type = RunApp
+    input = mesh.i
+    cli_args = '--mesh-only mesh.e'
+    recover = False # No recover for --mesh-only
+    executable_pattern = 'dire_wolf*'
```
Comment (GiudGiud) [inline microreactors/hpmr_assembly/tests:7]: can sockeye not run these inputs?

rn you re failing direwolf tests. You d need to update the DW submodule

Comment (joshuahansel) [reply microreactors/hpmr_assembly/tests:7]: It's using a material that's in Bison. And yeah, I was waiting for a DireWolf merge at the time - I'll update now.

**Comment by GiudGiud** [comment]
ugh we're being hit with direwolf treating warnings as errors. I need to debug this

**Comment by GiudGiud** [comment]
@joshuahansel

once the csvdiff on this new model is fixed it can go in

**Comment by GiudGiud** [comment]
oh CSV diff could be my fault from changing the transfer to the non-deprecated one..

**Comment by GiudGiud** [comment]
if you investigate before I do, 99% change it s just a bounding box issue

**Comment by GiudGiud** [comment]
htting an assertion on the mesh
```
*** ERROR ***
Assertion `factor != static_cast<T>(0.)' failed.
factor = 0
static_cast<T>(0.) = 0


Stack frames: 14
0: 0   libmesh_devel.0.dylib               0x0000000110ec7858 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char>>&) + 1044
1: 1   libmesh_devel.0.dylib               0x0000000110ec4628 libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*, std::__1::basic_ostream<char, std::__1::char_traits<char>>&) + 272
2: 2   libmoose-devel.0.dylib              0x000000010e6cf458 libMesh::TypeVector<double>::operator/=(double const&) + 560
3: 3   libmoose-devel.0.dylib              0x000000010f58fe88 MooseMeshUtils::meshCentroidCalculator(libMesh::MeshBase const&) + 1156
```

**Comment by joshuahansel** [comment]
I tried updating the transfer to the non-deprecated one, and the gold matches the deprecated one. Not sure what parameters you used. Maybe your bounding box was too small - did you use `error_on_miss = true` to be sure?

File: doc/content/microreactors/hpmr_assembly/index.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
+# Heat-Pipe Microreactor Assembly
+
+*Contact: Joshua Hansel (joshua.hansel.at.inl.gov)*
+
+*Model link: [Heat-Pipe Microreactor Assembly](https://github.com/idaholab/virtual_test_bed/tree/devel/microreactors/hpmr_assembly)*
+
```
Comment (GiudGiud) [inline doc/content/microreactors/hpmr_assembly/index.md:6]:
```suggestion

!tag name=Heat-Pipe Microreactor Assembly pairs=reactor_type:microreactor
                       reactor:HPMR
                       geometry:assembly
                       simulation_type:multiphysics
                       input_features:multiapps;reactor_meshing
                       transient:steady_state
                       codes_used:DireWolf;BISON;Sockeye
                       computing_needs:Workstation
                       fiscal_year:2024

```

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
I must have fixed a bug in the transfers since then.

**Pull request merged**